drivers/SmartThings/zigbee-switch/src/zigbee-dual-metering-switch/init.lua

```diff
@@ -25,31 +35,56 @@ local function can_handle_zigbee_dual_metering_switch(opts, driver, device, ...)
   return false
 end
 
-local function endpoint_to_component(device, endpoint)
-  if endpoint == 2 then
-    return "switch1"
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    local name = "AURORA Outlet 2"
+    local metadata = {
+      type = "EDGE_CHILD",
+      label = name,
+      profile = "switch-power-2",
+      parent_device_id = device.id,
+      parent_assigned_child_key = string.format("%02X", ENDPOINTS.child),
+      vendor_provided_label = name,
+    }
+    driver:try_create_device(metadata)
+  end
+end
+
+local function find_child(parent, ep_id)
+  if ep_id == 1 then
+    return parent
   else
-    return "main"
+    return parent:get_child_by_parent_assigned_key(string.format("%02X", ep_id))
   end
 end
 
 local function component_to_endpoint(device, component)
-  if component == "switch1" then
-    return 2
-  else
-    return 1
+  return ENDPOINTS.parent -- in zigbee drivers we are mapping components to numbers, not to array of numbers as in zwave cases
+end
+
+local function device_init(driver, device)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    device:set_find_child(find_child)
+    device:set_component_to_endpoint_fn(component_to_endpoint)
   end
 end
 
-local function map_components(self, device)
-  device:set_endpoint_to_component_fn(endpoint_to_component)
-  device:set_component_to_endpoint_fn(component_to_endpoint)
+local function do_refresh(self, device)
+  device:send(OnOff.attributes.OnOff:read(device))
+  device:send(SimpleMetering.attributes.Divisor:read(device))
+  device:send(SimpleMetering.attributes.Multiplier:read(device))
```

Comment:
@NikitaKuchta don't You need to read power meter attributes too ? (from all endpoints)

Reply:
done